### PR TITLE
OPSEXP-2147 Drop helper_modules role

### PR DESCRIPTION
**Pull request by gionn**
OPSEXP-2147

* Network checks playbook moved outside the role which was not really reusable
* Created docker scenario for quick testing
* Added into clustered scenario for proper e2e testing

### CODE REVIEW

File: playbooks/prerun-network-checks.yml [hunk 1]
```diff
@@ -1,188 +1,199 @@
...
   tasks:
+    - name: Include common defaults
+      ansible.builtin.include_vars: ../roles/common/defaults/main.yml
+
     - name: Check db connection
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.database | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.database[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+        checked_host: "{{ hostvars[groups.database[0]].ansible_host | default('localhost') }}"
```
Comment (alxgomz) [inline playbooks/prerun-network-checks.yml:14]: actually we should probablyuse `{{ db_host }}` as that's what's used in the repo role to connect to the db
https://github.com/Alfresco/alfresco-ansible-deployment/blob/fe23ffa25bea1ee545a69f57b8f555e88d2e03c4/roles/common/defaults/main.yml#L43

File: playbooks/prerun-network-checks.yml [hunk 2]
```diff
@@ -1,188 +1,199 @@
...
         delegate_target: "{{ groups.database | first }}"
-      when: repo_db_url == ""
+      when: repo_db_url == "" and groups.database | default([]) | length > 0
 
     - name: Check activemq connection
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.activemq | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.activemq[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+        checked_host: "{{ hostvars[groups.activemq[0]].ansible_host | default('localhost') }}"
```
Comment (alxgomz) [inline playbooks/prerun-network-checks.yml:22]: Same as above: `activemq_host`
https://github.com/Alfresco/alfresco-ansible-deployment/blob/fe23ffa25bea1ee545a69f57b8f555e88d2e03c4/roles/common/defaults/main.yml#L31

File: playbooks/prerun-network-checks.yml [hunk 3]
```diff
@@ -1,188 +1,199 @@
...
+        checked_host: "{{ hostvars[groups.search[0]].ansible_host | default('localhost') }}"
         checked_port: "{{ ports_cfg.search.http }}"
         delegate_target: "{{ groups.search | first }}"
       when: groups.search | default([]) | length > 0
 
     - name: Check sync connection
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.syncservice | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.syncservice[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
```
Comment (alxgomz) [inline playbooks/prerun-network-checks.yml:35]: `sync_host`
https://github.com/Alfresco/alfresco-ansible-deployment/blob/fe23ffa25bea1ee545a69f57b8f555e88d2e03c4/roles/common/defaults/main.yml#L85

File: playbooks/prerun-network-checks.yml [hunk 4]
```diff
@@ -1,188 +1,199 @@
...
         checked_port: "{{ ports_cfg.activemq[activemq_protocol] }}"
         delegate_target: "{{ groups.activemq | first }}"
       when: groups.activemq | default([]) | length > 0
 
     - name: Check search connection
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.search | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.search[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+        checked_host: "{{ hostvars[groups.search[0]].ansible_host | default('localhost') }}"
```
Comment (alxgomz) [inline playbooks/prerun-network-checks.yml:30]: `solr_host`
https://github.com/Alfresco/alfresco-ansible-deployment/blob/fe23ffa25bea1ee545a69f57b8f555e88d2e03c4/roles/common/defaults/main.yml#L49C1-L49C10

File: playbooks/prerun-network-checks.yml [hunk 5]
```diff
@@ -1,188 +1,199 @@
...
       when:
         - groups.syncservice | default([]) | length > 0
-        - acs.edition == "Enterprise"
 
     - name: Check sfs connection
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.transformers | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.transformers[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+        checked_host: "{{ hostvars[groups.transformers[0]].ansible_host | default('localhost') }}"
```
Comment (alxgomz) [inline playbooks/prerun-network-checks.yml:47]: `sfs_host`
https://github.com/Alfresco/alfresco-ansible-deployment/blob/fe23ffa25bea1ee545a69f57b8f555e88d2e03c4/roles/common/defaults/main.yml#L61

File: playbooks/prerun-network-checks.yml [hunk 6]
```diff
@@ -1,188 +1,199 @@
...
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.syncservice | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.syncservice[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+        checked_host: "{{ hostvars[groups.syncservice[0]].ansible_host | default('localhost') }}"
         checked_port: "{{ ports_cfg.sync.http }}"
         delegate_target: "{{ groups.syncservice | first }}"
       when:
         - groups.syncservice | default([]) | length > 0
-        - acs.edition == "Enterprise"
```
Comment (alxgomz) [inline playbooks/prerun-network-checks.yml:40]: Why are you excluding the enterprise ersion condition for sync and adding it for ats?

File: playbooks/prerun-network-checks.yml [hunk 7]
```diff
@@ -1,188 +1,199 @@
...
         delegate_target: "{{ groups.transformers | first }}"
-      when: acs.edition == "Enterprise"
+      when: acs.edition == "Enterprise" and groups.transformers | default([]) | length > 0
 
     - name: Check trouter connection
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.transformers | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.transformers[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+        checked_host: "{{ hostvars[groups.transformers[0]].ansible_host | default('localhost') }}"
```
Comment (alxgomz) [inline playbooks/prerun-network-checks.yml:55]: `trouter_host`
https://github.com/Alfresco/alfresco-ansible-deployment/blob/fe23ffa25bea1ee545a69f57b8f555e88d2e03c4/roles/common/defaults/main.yml#L67

File: playbooks/prerun-network-checks.yml [hunk 8]
```diff
@@ -1,188 +1,199 @@
...
         delegate_target: "{{ groups.transformers | first }}"
-      when: acs.edition == "Enterprise"
+      when: acs.edition == "Enterprise" and groups.transformers | default([]) | length > 0
 
     - name: Check tengine connection
-      ansible.builtin.include_tasks: "../roles/helper_modules/tasks/check_port.yml"
+      ansible.builtin.include_tasks: "tasks/check_port.yml"
       vars:
-        checked_host: "{% if groups.transformers | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.transformers[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
+        checked_host: "{{ hostvars[groups.transformers[0]].ansible_host | default('localhost') }}"
```
Comment (alxgomz) [inline playbooks/prerun-network-checks.yml:63]: `ats_tengine_aio_host`
https://github.com/Alfresco/alfresco-ansible-deployment/blob/fe23ffa25bea1ee545a69f57b8f555e88d2e03c4/roles/common/defaults/main.yml#L55C1-L55C21

**Review by alxgomz — CHANGES_REQUESTED** [review]

File: molecule/prerun_network_checks/host_vars/instance.yml [hunk 9]
```diff
@@ -0,0 +1,5 @@
+ansible_user: ansible
+# BEGIN KNOWN_URLS VAR
+known_urls:
+  - https://localhost/share/
+# END KNOWN_URLS VAR
```
Comment (gionn) [inline molecule/prerun_network_checks/host_vars/instance.yml:5]:
```suggestion
```

File: playbooks/tasks/check_port.yml [hunk 10]
```diff
@@ -1,8 +1,7 @@
-  - name: Include common defaults
-    ansible.builtin.include_vars: ../../common/defaults/main.yml
-
-  - name: Check connectivity
-    block:
+- name: Check connectivity
+  tags:
+    - molecule-idempotence-notest
```
Comment (gionn) [inline playbooks/tasks/check_port.yml:3]: listen is not idempotent so skipping everything

**Review by pmacius — APPROVED** [review]
Did't catch anything - looks good

**Review by alxgomz — APPROVED** [review]

**Pull request merged**